Task/clean code vm text

gRaytan opened the pull request.

teamagam3 committed:
Finished GeoDataRepository

Added basic models to work with
Not sure about InMemoryGeoDataCache, currently not in use

teamagam3 committed:
Moved logic to InMemoryGeoDataCache

teamagam1 committed:
basic viewmodel for map with data binding

teamagam3 committed:
Added basic interactor

teamagam3 committed:
Added basic classes

teamagam3 committed:
Moved classes

teamagam3 committed:
Fixed package name

teamagam3 committed:
Finished GeometryInteractor

Added basic presenter for geo message

teamagam3 committed:
Finished SendGeoEntityPresenter implementation

There is still a implementation problem with the observable, we have to object to return (one is the message and the other one is the GeoEntity), we have to think about it and decide what to return

teamagam1 committed:
added map component and module.
viewer fragment cleaned from VM code

teamagam3 committed:
MessageGeo model now contains the GeoEntity

teamagam1 committed:
injection fixes

teamagam1 committed:
model of map moved to map directory

teamagam3 committed:
Fixed build errors

teamagam1 committed:
view (viewer) of map moved to map directory

teamagam1 committed:
retrolambda removed from app for java 7 competability.

teamagam3 committed:
Added dependency injection

teamagam1 committed:
CR fixes

teamagam3 committed:
Merge pull request #101 from teamagam/task/refactor_geo_message/vm_for_viewer_fragment

task/refactor_geo_message/vm_for_viewer_fragment

teamagam3 committed:
Changed to code of adding VectorLayer[] to the repository

teamagam1 committed:
gestures moved to vm

teamagam3 committed:
Added ImagesDataRepository

teamagam3 committed:
CR fixes

teamagam3 committed:
Merge pull request #99 from teamagam/task/vector_layer_repository

Added VectorLayer repository.

teamagam3 committed:
CR fixes

Define the symbol interface task still remains until further discussion

teamagam1 committed:
basic data binding

teamagam3 committed:
CR fixes

Added image compression

teamagam3 committed:
Added basic classes

teamagam1 committed:
Merge pull request #100 from teamagam/task/geo_presenter_interactor

Finished implementing GeoInteractor and GeoPresenter.

teamagam1 committed:
Repository path fix

teamagam3 committed:
CR fixes

teamagam3 committed:
Merge pull request #102 from teamagam/task/image_repository

Finished ImagesRepository.

teamagam3 committed:
Finished SendImageMessageInteractor and SendMessageImagePresenter

teamagam3 committed:
Removed Nullable annotaion

gRaytan committed:
start implemneting VM for SendTextMessage

teamagam3 committed:
Added basic class

gRaytan committed:
finish vm for SendMessageDialogFragment.java

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/104" target="_blank"><img src="https://app.review.ninja/52010471/pull/104/badge" alt="ReviewNinja"/>

teamagam3 committed:
Added basic logic of taking image with VM

teamagam3 committed:
Moved the compressing image functionality to ImageDataReposistory

teamagam3 committed:
Added logger on error

teamagam3 committed:
Changed ImageRepository to return Observable of MessageImage

teamagam3 committed:
buildUseCaseObservable of SendImageMessageInteractor now directly uses the repository methods

teamagam3 committed:
Fixed the image flow and errors

teamagam3 committed:
Removed static sender id

teamagam1 committed:
view model of send geo message works with data binding and dependency injection

teamagam1 committed:
butterknife fix

teamagam1 committed:
deleted unused code in vm

teamagam3 committed:
Renamed the ViewModel class name

Added databinding to gradle

teamagam1 committed:
fixed two way data binding of spinner

teamagam3 commented on app/build.gradle:
```diff
@@ -14,6 +14,10 @@ android {
             storeFile file('../keystores/gimelgimel.jks')
         }
     }
+    dexOptions {
```
Unnecessary for all of us, should be used locally

teamagam3 commented on app/src/main/java/com/teamagam/gimelgimel/app/viewModels/SendMessageDialogFragmentViewModel.java:
```diff
@@ -0,0 +1,106 @@
+    private boolean isInputValid(String userMessage) {
+        return !userMessage.isEmpty();
+    }
+    private class handlePresenterView implements SendMessagePresenter.View {
```
The view should implements these methods.
The view model shouldn't do UI tasks like toast.

teamagam3 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendMessageDialogFragment.java:
```diff
@@ -105,41 +81,19 @@ protected Object castInterface(Fragment fragment) {
+                .inject(this);
     }
 
-    private class handlePresenterView implements SendMessagePresenter.View{
```
Restore this class (Already explained why).

teamagam3 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendMessageDialogFragment.java:
```diff
@@ -2,67 +2,43 @@
-
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mBinding = DataBindingUtil.inflate(LayoutInflater.from(getActivity()), R.layout.dialog_send_message, null, false);
```
What's the difference between DataBindingUtil.inflate() and DataBindingUtils.setContentView()?

there is no use of setContentView

teamagam3 commented on app/src/main/java/com/teamagam/gimelgimel/app/viewModels/SendMessageDialogFragmentViewModel.java:
```diff
@@ -0,0 +1,106 @@
+    }
+    private class handlePresenterView implements SendMessagePresenter.View {
+
+        final Context mContext;
```
Context is part of the view, this is another reason why this class has to be moved to the fragment.

right. no need

teamagam3 commented on app/src/main/java/com/teamagam/gimelgimel/app/viewModels/SendMessageDialogFragmentViewModel.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+
+    public boolean onPositiveClick() {
```
Rename the method to better name that describes its goal and move it to the ViewModel (The dialog has this method too).

teamagam3 commented on app/src/main/java/com/teamagam/gimelgimel/app/viewModels/ViewModel.java:
```diff
@@ -0,0 +1,10 @@
+ */
+public interface ViewModel {
+
+    void destroy();
```
Why only destroy()?

There is no common functionality yet. we will edit after merge and other UC

teamagam3 commented on app/src/main/res/layout/dialog_send_message.xml:
```diff
@@ -1,15 +1,26 @@
-        android:inputType="text" />
-</LinearLayout>
\ No newline at end of file
+    <data>
```
The data on binding is the model the view works with.
The view model is just another bridge between the model and the view that updated the model by the view actions.

Didnt understand your comment

teamagam3 replied:
Nevermind, we talked about our architecture, and the binding data will be the view model.

teamagam3 commented on xwalk_core_library/build.gradle:
```diff
@@ -8,7 +8,10 @@ android {
-
+    dexOptions {
+        // Prevent OutOfMemory with MultiDex during the build phase
+        javaMaxHeapSize "4g"
```
Same same.

teamagam3 commented on data/build.gradle:
```diff
@@ -12,7 +12,10 @@ targetCompatibility = 1.8
-
+    dexOptions {
+        // Prevent OutOfMemory with MultiDex during the build phase
+        javaMaxHeapSize "4g"
```
Same same.

teamagam1 committed:
bug fix of 20 characters with databinding

gRaytan committed:
fixing CR comments

teamagam3 committed:
Added dependency injection

teamagam3 committed:
CR fixes

teamagam3 committed:
Rebase fixes

teamagam3 committed:
Circle CI fix attempt

teamagam3 committed:
Added null check

teamagam3 committed:
Changed doOnComplete to doOnNext

getImagesBytes now returns Observable<byte[]>

teamagam3 committed:
Merge pull request #103 from teamagam/task/image_presenter_interactor

Finished SendImageInteractor and SendImagePresenter.

teamagam3 committed:
CR fixes

teamagam1 committed:
CR fixes:

viewModel of geoMessage holds a model.

teamagam1 committed:
dataBinding of text validation with editText of the sendGeoMessage

teamagam1 committed:
Merge pull request #105 from teamagam/task/refactor_geo_message/send_geo_vm

task/refactor_geo_message/send_geo_vm

teamagam1 committed:
Presenter with new architecture basic works

teamagam1 committed:
presenter fix for the new architecture.

need to fix: singelton presenter?

teamagam1 committed:
presenter is now part of presenters module which is singelton.

teamagam1 committed:
presenter injection to app-component (singleton)

teamagam3 committed:
Finished VM

teamagam3 committed:
Replaced toast with snackbar

teamagam3 committed:
Circle CI fix

teamagam1 committed:
CR fixes

gRaytan committed:
refactor

teamagam1 committed:
CR fixes. interactor - repository fixes

teamagam1 committed:
mapComponent merged with messagesComponent to mainActivity component

teamagam1 committed:
generify presenter with interface for views

teamagam3 committed:
Changed to new architecture

gRaytan committed:
fixes in text messages

barmako committed:
Send (Text) message use-case first block done.
The app sends text messages via SendMessageInteractor.
TBD: message sending result user - notification not implemented yet.

Fixed a bug for both text and geo message dialog fragments - update OK button's enabled status on fragment resume (it'll be disabled when the dialog is displayed at first now).

teamagam3 committed:
CR Fixes

barmako committed:
SendGeoMessageInteractor now creates the message by itself, instead of the ViewModel.
Forced interactors to declare their observable working types (and therefore, the requested subscriber's data-type)

barmako committed:
Changed interactors to work with exeute alone.
Added SendMessageInteractor as a base class for both SendGeoMessageInteractor and SendTextMesssageInteractor.
Added AutoFactory to allow assisted injection for the interactors.
Disabled MapViewModel update as a direct result of geo message sending - will be implemented by a separate interactor (TBD).
gradle.properties: allowed more RAM for compilation

barmako committed:
Merge remote-tracking branch 'origin/task/images_vm' into HEAD

# Conflicts:
#	app/src/main/java/com/teamagam/gimelgimel/app/GGApplication.java
#	app/src/main/java/com/teamagam/gimelgimel/app/injectors/components/ApplicationComponent.java
#	app/src/main/java/com/teamagam/gimelgimel/app/injectors/components/MessagesComponent.java
#	app/src/main/java/com/teamagam/gimelgimel/app/injectors/modules/ApplicationModule.java
#	app/src/main/java/com/teamagam/gimelgimel/app/injectors/modules/MessageModule.java
#	app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java
#	app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendMessageDialogFragment.java
#	app/src/main/res/layout/fragment_cesium.xml
#	data/src/main/java/com/teamagam/gimelgimel/data/message/adapters/MessageDataMapper.java
#	presentation/src/main/java/com/teamagam/gimelgimel/presentation/presenters/base/AbstractPresenter.java

barmako committed:
Merged send image message branch with geo and text message sending.
Refactored merged branch to work with common interactors.
